### ISSUE TITLE
Max current specification?

### ISSUE THREAD

**Issue by tonydemark** [standupmaths/xmastree2021#63]
Should we be trying to stay below a max current? I'm assuming each R, G, B element is 20 mA and calculating a max value. Undimmed one or two of my CSV files max out at 20A. I figured that was kinda excessive so I put in a brightness limiter that keeps them generally under 10A.

**Comment by d-albrecht** [comment]
Matt is already thinking about having a dimming factor to avoid power supply induced flickering. But we don't have conformation yet whether this will be available or not. Ideally, you don't have to care and should just use some (arbitrary) max-value that then can get scaled.

Some animations can clearly ignore this for a reason because the animation isn't addressing a huge number of lights at the same time. If you do, though, you can of course dim your animation precautionary. But then, don't dim too much. If you can't see your animation on any of the simulators, it's definitely too dim even for the tree.

If you watch the initial tree video you might be able to figure out what kind of LEDs he uses and what power they draw. I don't know and I don't really care. But the problematic thing also is that - if I remember correctly - it isn't just a max power draw overall, but individual limits for patches of lights. And someone somewhere in this repo mentioned that the power situation got improved compared to last year, but I don't know whether this is true and if so by how much.

Finally, there is the problem that I don't know how the timings work. I know for some electronics components that they can switch one way faster than the other. So, it might even be possible that you exceed the power limits by switching from one half of the lights to the other, because for a split second all color channels are on. Just saying, there a lot of variables if you wanted to really prevent any power issues here.

TL;DR: If your dimmed animation is still enjoyable on one of the simulators, go ahead and din them to be safe, but currently there is no rule that animations must be within some (undefined) limit.
